mdn/src/main/java/org/apache/james/mdn/MDNFactory.java

```diff
@@ -72,6 +72,28 @@ static public MimeMultipartReport create(String humanText,
         multiPart.addBodyPart(humanPart);
 
         // Part 2: MDN Report Part
+        String mdnReport = generateMDNReport(reporting_UA_name,
+            reporting_UA_product,
+            original_recipient,
+            final_recipient,
+            original_message_id,
+            disposition);
+
+        MimeBodyPart mdnPart = new MimeBodyPart();
+        mdnPart.setContent(mdnReport, "message/disposition-notification");
+        multiPart.addBodyPart(mdnPart);
```

Comment:
maybe split into different methods instead of comments part 1 / part 2 / part 3

Reply:
I need to extract this in a MDN object, as discussed with @mbaechler . Will Do.

Reply:
Done